libraries/botbuilder-dialogs-declarative/src/componentDeclarativeTypes.ts

```diff
@@ -15,3 +17,14 @@ import { ResourceExplorer } from './resources';
 export interface ComponentDeclarativeTypes {
     getDeclarativeTypes(resourceExplorer: ResourceExplorer): DeclarativeType[];
 }
+
+/**
+ * Check if a [ComponentRegistration](xref:botbuilder-core.ComponentRegistration) is
+ * [ComponentDeclarativeTypes](xref:botbuilder-dialogs-declarative.ComponentDeclarativeTypes) or not.
+ *
+ * @param {ComponentRegistration} component The component registration.
+ * @returns {boolean} Type check result.
+ */
+export function isComponentDeclarativeTypes(component: ComponentRegistration): component is ComponentDeclarativeTypes {
```

Comment:
It's more common to annotate the argument as `any` (I know, I'm always suggesting the opposite, but in this case, it makes sense to do the type checking).